lib/mirage.ml

```diff
@@ -1530,31 +1533,50 @@ let configure_makefile ~target ~root ~name ~warn_error info =
                   \t  -o mir-%s.virtio"
         extra_c_archives pkg_config_deps name ;
       append fmt "\t@@echo Build succeeded";
+      newline fmt;
+      append fmt "clean::\n\
+              \tocamlbuild -clean";
       R.ok ()
     | `Ukvm ->
       get_extra_ld_flags "freestanding" libs
       >>| String.concat ~sep:" \\\n\t  "
-      >>= fun extra_c_archives ->
-      append fmt "build:: main.native.o";
-      let pkg_config_deps = "mirage-solo5 ocaml-freestanding" in
-      append fmt "\tpkg-config --print-errors --exists %s" pkg_config_deps;
-      append fmt "\tld $$(pkg-config --variable=ldflags solo5-kernel-ukvm) \\\n\
+        >>= fun extra_c_archives ->
```

Comment:
I don't think this comment is specific to your change. But won't it be very hard to touch this mirage file when it gets upstreamed? wouldn't it be better to move this solo5 logic to some solo5 specific repo (somehow)?